### ISSUE TITLE
[Bug]: Metamask Extension Stuck on Loading State

### ISSUE THREAD

**Issue by prohimone** [MetaMask/metamask-extension#23323]
### Describe the bug

Metamask stuck on loading state on startup (similar to during issue [#22355](https://github.com/MetaMask/metamask-extension/issues/22355)) this time on brave and I get an error below:

```
TypeError: globalThis.stateHooks.getPersistedState is not a function
  at i.o [as getMetaMetricsEnabled] (sentry-install.js:1:3225)
  at sentry-install.js:1:2377
  at sentry-install.js:1:128163
  at new i (sentry-install.js:1:359558)
  at i._notifyEventProcessors (sentry-install.js:1:128071)
  at sentry-install.js:1:128399
  at new i (sentry-install.js:1:359558)
  at i._notifyEventProcessors (sentry-install.js:1:128071)
  at sentry-install.js:1:128399
  at new i (sentry-install.js:1:359558)
  at i._notifyEventProcessors (sentry-install.js:1:128071)
  at sentry-install.js:1:128399
  at new i (sentry-install.js:1:359558)
  at i._notifyEventProcessors (sentry-install.js:1:128071)
  at sentry-install.js:1:128399
  at new i (sentry-install.js:1:359558)
  at i._notifyEventProcessors (sentry-install.js:1:128071)
  at i.applyToEvent (sentry-install.js:1:127862)
  at n.prepareEvent (sentry-install.js:1:157348)
  at u._prepareEvent (sentry-install.js:1:103677)
  at u._prepareEvent (sentry-install.js:1:56531)
  at u._processEvent (sentry-install.js:1:104472)
  at u._captureEvent (sentry-install.js:1:103735)
  at u.captureEvent (sentry-install.js:1:100833)
  at sentry-install.js:1:110904
  at u._withClient (sentry-install.js:1:113203)
  at u.captureEvent (sentry-install.js:1:110881)
  at d (sentry-install.js:1:71844)
  at sentry-install.js:1:70873
  at h (sentry-install.js:1:338033)
  at l.onerror (sentry-install.js:1:337477)
  ```
  
### Expected behavior
No response

### Screenshots/Recordings
No Response 

### Steps to reproduce
1. Click the metamask extension
2. Stuck on first load

### Version
11.7.2

### Build type
None

### Browser
Brave

### Operating system
MacOS

### Hardware wallet
No response

### Additional context
No response

### Severity
No response

**Comment by shazow** [comment]
A lot of recent reviews agree: 
![image](https://github.com/MetaMask/metamask-extension/assets/6292/81becf3c-5f53-4280-b4e6-4ba8638b94fe)

**Comment by cryptodigi** [comment]
same issues on 11.11.4

**Comment by cryptodigi** [comment]
error still present, not able to get to password prompt, always loading

**Comment by GeorgePapageorgakis** [comment]
Same for Firefox. Is this bug on purpose to lock people out of their funds? What is going on? Time to switch wallets...

**Comment by gauthierpetetin** [comment]
Still present in 11.11.4: https://github.com/MetaMask/metamask-extension/issues/23409

**Comment by cryptodigi** [comment]
still loading forever on 11.12.0

**Comment by cryptodigi** [comment]
SES_UNHANDLED_REJECTION: (TypeError#2)

still get this error on 11.12.1

**Comment by cryptodigi** [comment]
still loading forever on 11.12.4

**Comment by gauthierpetetin** [comment]
Hi everyone, if you're still facing a similar issue, this issue is the possible cause: https://github.com/MetaMask/metamask-extension/issues/9196

**Comment by cryptodigi** [comment]
is this a joke????

**Comment by gauthierpetetin** [comment]
Hi @cryptodigi , just to clarify on the above, I updated the status from "Closed as completed" into "Closed as not planned", because:
- "Closed as completed" was not accurate since the bug isn't solved, we're still trying to solve it
- "Closed as not planned" is the status to use for `duplicate` issues, which is accurate in this case
In conclusion, we're trying to fix this issue (https://github.com/MetaMask/metamask-extension/issues/9196) which we believe has the same root cause

**Comment by cryptodigi** [comment]
> Hi @cryptodigi , Just to clarify on the above, I updated the status from "Closed as completed" into "Closed as not planned", because:
> 
> * "Closed as completed" was not accurate since the bug isn't solved, we're still trying to solve it
> * "Closed as not planned" is the status to use for `duplicate` issues, which is accurate in this case
> 
> Indeed, we're trying to fix this issue (#9196) which we believe has the same root cause

cool thanks
